monai/fl/client/monai_algo.py

```diff
@@ -435,71 +418,58 @@ def initialize(self, extra=None):
-        self.eval_parser[RequiredBundleKeys.BUNDLE_ROOT] = self.bundle_root
-        # number of training epochs for each round
-        if BundleKeys.TRAIN_TRAINER_MAX_EPOCHS in self.train_parser:
-            self.train_parser[BundleKeys.TRAIN_TRAINER_MAX_EPOCHS] = self.local_epochs
-
-        # remove checkpoint loaders
-        if self.disable_ckpt_loading:
-            disable_ckpt_loaders(self.train_parser)
-            disable_ckpt_loaders(self.eval_parser)
-
         # set tracking configs for experiment management
         if self.tracking is not None:
             if isinstance(self.tracking, str) and self.tracking in DEFAULT_EXP_MGMT_SETTINGS:
                 settings_ = DEFAULT_EXP_MGMT_SETTINGS[self.tracking]
             else:
                 settings_ = ConfigParser.load_config_files(self.tracking)
-            ConfigWorkflow.patch_bundle_tracking(parser=self.train_parser, settings=settings_)
-            ConfigWorkflow.patch_bundle_tracking(parser=self.eval_parser, settings=settings_)
 
-        # Get trainer, evaluator
-        self.trainer = self.train_parser.get_parsed_content(
-            BundleKeys.TRAINER, default=ConfigItem(None, BundleKeys.TRAINER)
-        )
-        self.evaluator = self.eval_parser.get_parsed_content(
-            BundleKeys.EVALUATOR, default=ConfigItem(None, BundleKeys.EVALUATOR)
-        )
+        if self.train_workflow is None and self.config_train_filename is not None:
+            config_train_files = self._add_config_files(self.config_train_filename)
+            self.train_workflow = ConfigWorkflow(
+                config_file=config_train_files, meta_file=None, logging_file=None, workflow="train"
+            )
+        if self.train_workflow is not None:
+            self.train_workflow.initialize()
+            self.train_workflow.bundle_root = self.bundle_root
+            self.train_workflow.max_epochs = self.local_epochs
+            if self.tracking is not None and isinstance(self.train_workflow, ConfigWorkflow):
+                ConfigWorkflow.patch_bundle_tracking(parser=self.train_workflow.parser, settings=settings_)
+            if self.disable_ckpt_loading and isinstance(self.train_workflow, ConfigWorkflow):
+                disable_ckpt_loaders(parser=self.train_workflow.parser)
+            # initialize the workflow as the content changed
+            self.train_workflow.initialize()
+            self.trainer = self.train_workflow.trainer
+            if not isinstance(self.trainer, SupervisedTrainer):
+                raise ValueError(f"trainer must be SupervisedTrainer, but got: {type(self.trainer)}.")
+
+        if self.eval_workflow is None and self.config_evaluate_filename is not None:
+            config_eval_files = self._add_config_files(self.config_evaluate_filename)
+            self.eval_workflow = ConfigWorkflow(
+                config_file=config_eval_files, meta_file=None, logging_file=None, workflow="train"
```

Comment:
Should this be `workflow="eval"`?

Reply:
The workflow is "train" because all the model-zoo bundles override "train.json" workflow with a "evaluate.json" config.
I already updated the "config_fl_evaluate.json" to override "config_fl_train.json" to simulate real world bundles in the latest commit.

Thanks.

Reply:
And I also added new arg "eval_workflow_name" to set it.

Thanks.